components/portal/messages/controllers.js

```diff
@@ -95,20 +100,18 @@ define(['angular'], function(angular) {
          * @param {Object} result
          */
         var filterMessagesSuccess = function(result) {
-          // Check for filtered notifications
-          var filteredMessages = [];
-          if (result.filteredByGroup && result.filteredByData) {
-            // Combine the two filtered arrays into one (no dupes)
-            filteredMessages = $filter('filterForCommonElements')(
-              result.filteredByGroup,
-              result.filteredByData
-            );
-            var dateFilteredMessages =
-              $filter('filterByDate')(filteredMessages);
-            $scope.messages =
-              $filter('separateMessageTypes')(dateFilteredMessages);
-            $scope.hasMessages = true;
-          }
+          $scope.seenMessageIds = result[0];
+          var dataMessages = result[1];
+          var groupedMessages = result[2];
+          var filteredMessages = $filter('filterForCommonElements')(
+            groupedMessages,
+            dataMessages
+          );
+          var dateFilteredMessages =
+            $filter('filterByDate')(filteredMessages);
+             $scope.messages =
```

Comment:
Not a big deal, but the indentation of this line confused me quite a bit.  It took a while for me to figure out that line 112 was not part of the datFilteredMessages function.